common/persistence/src/test/java/app/coronawarn/server/common/persistence/service/DiagnosisKeyServiceTest.java

```diff
@@ -149,15 +150,39 @@ void testNoPersistOnValidationError() {
     assertDiagnosisKeysEqual(Lists.emptyList(), actKeys);
   }
 
-  public static DiagnosisKey buildDiagnosisKeyForSubmissionTimestamp(long submissionTimeStamp) {
+  @Test
+  void shouldNotUpdateExistingKey() {
+    var keys = List.of(DiagnosisKey.builder()
+            .withKeyData("1234567890123456".getBytes())
+            .withRollingStartIntervalNumber((int) (OffsetDateTime.now(UTC).toEpochSecond() / 600) - 2)
+            .withTransmissionRiskLevel(2)
+            .withSubmissionTimestamp(0L).build(),
+        DiagnosisKey.builder()
+            .withKeyData("1234567890123456".getBytes())
+            .withRollingStartIntervalNumber((int) (OffsetDateTime.now(UTC).toEpochSecond() / 600) - 3)
+            .withTransmissionRiskLevel(3)
+            .withSubmissionTimestamp(0L).build());
+
+    diagnosisKeyService.saveDiagnosisKeys(keys);
+
+    var actKeys = diagnosisKeyService.getDiagnosisKeys();
+
+    assertThat(actKeys.size()).isEqualTo(1);
+    assertThat(actKeys.iterator().next().getTransmissionRiskLevel()).isEqualTo(2);
+  }
+
+  public DiagnosisKey buildDiagnosisKeyForSubmissionTimestamp(long submissionTimeStamp) {
+    byte[] randomBytes = new byte[16];
+    Random random = new Random(submissionTimeStamp);
```

Comment:
With regards to reproducible test results, should we consider not generating random input values for tests?

Reply:
We need different diagnosis keys here, since the second one wouldn't be saved otherwise, due to the new constraints on the db. This whole random thing, isn't really random. Having the same random seed (submission timestamp in this case) will result in the same bytes being generated.